big-o-notation/big-o.js

```diff
@@ -0,0 +1,47 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {}; // 1 * 1 = O(1)
+  const unique = []; // 1 * 1 = O(1)
+  for (
+    let i = 0; // 1 * 1 = O(1)
+    i < words.length; // 2 * n = O(n)
+    i++ // 2 * n = O(n)
+  ) {
+    const word = words[i]; // 2 * n = O(n)
+    if (!seen[word]) {
+      // 2 * n = O(n^2)
```

Comment:
This is execute just once each time through the loop, so it should be O(n). That is, it executes once for each of the `n` elements in the `words` array.

Same for the following two lines.

Oh, it looks like you mistakenly went from `2 * n` to `O(n^2)`. You should instead throw away the coefficient (2) and keep just the `n`, so it's O(n). The same mistake was made down below, too.